src/WebJobs.Script/Host/Kubernetes/KubernetesClient.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+
+namespace Microsoft.Azure.WebJobs.Script
+{
+    internal class KubernetesClient
+    {
+        private const int LeaseRenewDeadline = 10;
+        private readonly HttpClient _httpClient;
+        private readonly string _httpLeaderEndpoint;
+
+        internal KubernetesClient(IEnvironment environment, HttpClient httpClient = null)
+        {
+            _httpClient = httpClient ?? new HttpClient();
+            _httpLeaderEndpoint = environment.GetHttpLeaderEndpoint();
+        }
+
+        internal async Task<KubernetesLockHandle> GetLock(string lockName, CancellationToken cancellationToken)
+        {
+            if (string.IsNullOrEmpty(lockName))
+            {
+                throw new ArgumentNullException(nameof(lockName));
+            }
+
+            var request = new HttpRequestMessage()
+            {
+                Method = HttpMethod.Get,
+                RequestUri = GetRequestUri($"?name={lockName}")
+            };
+
+            var response = await _httpClient.SendAsync(request, cancellationToken);
+
+            response.EnsureSuccessStatusCode();
+
+            var responseString = await response.Content.ReadAsStringAsync();
+            return JsonConvert.DeserializeObject<KubernetesLockHandle>(responseString);
+        }
+
+        internal async Task<KubernetesLockHandle> TryAcquireLock (string lockId, string ownerId, TimeSpan lockPeriod, CancellationToken cancellationToken)
+        {
+            if (string.IsNullOrEmpty(lockId))
+            {
+                throw new ArgumentNullException(nameof(lockId));
+            }
+
+            if (string.IsNullOrEmpty(ownerId))
+            {
+                throw new ArgumentNullException(nameof(ownerId));
+            }
```

Comment:
validation for lockPeriod? establish a min lock period?

Reply:
This comes from the webjobs sdk. Lock period is not validated in the storage lock manager as well.